cartography/rules/runners.py

```diff
@@ -0,0 +1,264 @@
+        total_facts=total_facts,
+        total_findings=total_findings,
+    )
+
+
+def _format_and_output_results(
+    all_results: list[FrameworkResult],
+    framework_names: list[str],
+    output_format: str,
+    total_requirements: int,
+    total_facts: int,
+    total_findings: int,
+):
+    """Format and output the results of framework execution."""
+    if output_format == "json":
+        if len(framework_names) == 1:
+            print(json.dumps(asdict(all_results[0]), indent=2))
+        else:
+            combined_output = [asdict(result) for result in all_results]
+            print(json.dumps(combined_output, indent=2))
+    else:
+        # Text summary
+        print("\n" + "=" * 60)
+        if len(framework_names) == 1:
+            print(f"EXECUTION SUMMARY - {FRAMEWORKS[framework_names[0]].name}")
+        else:
+            print("OVERALL SUMMARY")
+        print("=" * 60)
+
+        if len(framework_names) > 1:
+            print(f"Frameworks executed: {len(framework_names)}")
+        print(f"Requirements: {total_requirements}")
+        print(f"Total facts: {total_facts}")
+        print(f"Total results: {total_findings}")
+
+        if total_findings > 0:
+            print(
+                f"\n\033[36mFramework execution completed with {total_findings} total results\033[0m"
+            )
+        else:
+            print("\n\033[90mFramework execution completed with no results\033[0m")
+
+
+def run_frameworks(
+    framework_names: list[str],
+    uri: str,
+    neo4j_user: str,
+    neo4j_password: str,
+    neo4j_database: str,
+    output_format: str = "text",
+):
+    """
+    Execute the specified frameworks and present results.
+
+    :param framework_names: The names of the frameworks to execute.
+    :param uri: The URI of the Neo4j database. E.g. "bolt://localhost:7687" or "neo4j+s://tenant123.databases.neo4j.io:7687"
+    :param neo4j_user: The username for the Neo4j database.
+    :param neo4j_password: The password for the Neo4j database.
+    :param neo4j_database: The name of the Neo4j database.
+    :param output_format: Either "text" or "json". Defaults to "text".
+    :return: The exit code.
+    """
+    # Validate all frameworks exist
+    for framework_name in framework_names:
+        if framework_name not in FRAMEWORKS:
+            if output_format == "text":
+                print(f"Unknown framework: {framework_name}")
+                print(f"Available frameworks: {', '.join(FRAMEWORKS.keys())}")
+            return 1
+
+    # Connect to Neo4j
+    if output_format == "text":
+        print(f"Connecting to Neo4j at {uri}...")
+    driver = GraphDatabase.driver(uri, auth=(neo4j_user, neo4j_password))
+
+    try:
+        # Test connection
+        with driver.session(database=neo4j_database) as session:
+            result = session.run("RETURN 1 as test")
+            if result.single()["test"] != 1:
+                raise Exception("Connection test failed")
```

Comment:
There's a utility for this: `driver.verify_connectivity()`
https://neo4j.com/docs/python-manual/current/connect/